js/src/Ice/Reference.js

```diff
@@ -66,36 +67,36 @@ export class ReferenceFactory {
         this._defaultLocator = null;
     }
 
-    create(ident, facet, tmpl, endpoints) {
+    create(ident, facet, template, endpoints) {
```

Comment:
I would move ReferenceFactory to its own file, like in other languages. Also, remove the comment "Only for use by Instance" - that's no longer true.

Reply:
I would do it in a follow-up PR.